src/client/Microsoft.Identity.Client/Platforms/Shared/Apple/AppleDeviceAuthManager.cs

```diff
@@ -4,14 +4,15 @@
 using System.Collections.Generic;
 using System.Globalization;
 using System.Threading.Tasks;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
 
 namespace Microsoft.Identity.Client.Platforms.Shared.Apple
 {
-    internal class DeviceAuthHelper
+    internal class AppleDeviceAuthManager : IDeviceAuthManager
     {
-        public static bool CanHandleDeviceAuthChallenge { get { return false; } }
+        public bool CanHandleDeviceAuthChallenge { get { return false; } }
 
-        public static Task<string> CreateDeviceAuthChallengeResponseAsync(IDictionary<string, string> challengeData)
+        public Task<string> CreateDeviceAuthChallengeResponseAsync(IDictionary<string, string> challengeData)
```

Comment:
Why doesn't this method just throw or do nothing since CanHandle.. returns false..

Reply:
There are some places in code that call the CreateDeviceAuthChallengeResponseAsync directly without checking CanHandle for iOS and android. The response returned here tells AAD to bypass the PKeyAuth check.